test/schema-feature.test.js

```diff
@@ -281,61 +276,82 @@ test('Should throw if the schema headers is undefined', t => {
     }
   })
 
-  fastify.ready(err => {
-    t.equal(err.code, 'FST_ERR_SCH_VALIDATION_BUILD')
-    t.equal(err.message, 'Failed building the validation schema for GET: /:id, due to error headers schema is undefined')
-  })
+  fastify.ready()
 })
 
-test('Should throw if the schema params is undefined', t => {
+test('Should emit warning if the schema body is undefined', t => {
   t.plan(2)
   const fastify = Fastify()
 
+  process.on('warning', onWarning)
+  function onWarning (warning) {
+    t.equal(warning.name, 'FastifyDeprecation')
+    t.equal(warning.code, 'FSTDEP015')
+  }
+
+  t.teardown(() => {
+    process.removeListener('warning', onWarning)
+    warning.emitted.set('FSTDEP015', false)
+  })
+
   fastify.get('/:id', {
     handler: echoParams,
     schema: {
-      params: undefined
+      body: undefined
     }
   })
 
-  fastify.ready(err => {
-    t.equal(err.code, 'FST_ERR_SCH_VALIDATION_BUILD')
-    t.equal(err.message, 'Failed building the validation schema for GET: /:id, due to error params schema is undefined')
-  })
+  fastify.ready()
 })
 
-test('Should throw if the schema query is undefined', t => {
+test('Should emit warning if the schema query is undefined', t => {
   t.plan(2)
   const fastify = Fastify()
 
+  process.on('warning', onWarning)
+  function onWarning (warning) {
+    t.equal(warning.name, 'FastifyDeprecation')
+    t.equal(warning.code, 'FSTDEP015')
+  }
+
+  t.teardown(() => {
+    process.removeListener('warning', onWarning)
+    warning.emitted.set('FSTDEP015', false)
+  })
+
   fastify.get('/:id', {
     handler: echoParams,
     schema: {
       querystring: undefined
     }
   })
 
-  fastify.ready(err => {
-    t.equal(err.code, 'FST_ERR_SCH_VALIDATION_BUILD')
-    t.equal(err.message, 'Failed building the validation schema for GET: /:id, due to error querystring schema is undefined')
-  })
+  fastify.ready()
 })
 
-test('Should throw if the schema query is undefined', t => {
+test('Should emit warning if the schema params is undefined', t => {
   t.plan(2)
   const fastify = Fastify()
 
+  process.on('warning', onWarning)
+  function onWarning (warning) {
+    t.equal(warning.name, 'FastifyDeprecation')
+    t.equal(warning.code, 'FSTDEP015')
+  }
+
+  t.teardown(() => {
+    process.removeListener('warning', onWarning)
+    warning.emitted.set('FSTDEP015', false)
+  })
+
   fastify.get('/:id', {
     handler: echoParams,
     schema: {
-      querystring: undefined
+      params: undefined
     }
   })
 
-  fastify.ready(err => {
-    t.equal(err.code, 'FST_ERR_SCH_VALIDATION_BUILD')
-    t.equal(err.message, 'Failed building the validation schema for GET: /:id, due to error querystring schema is undefined')
-  })
+  fastify.ready()
```

Comment:
`Fastify#ready` returns a Promise 😅

Reply:
@metcoder95 I have two proposals. What do you think could be better for those specific tests?

1. Call `ready` with a callback
```js
fastify.ready(err => t.error(err))
```

2. Use inject and test the endpoint
```js
  fastify.inject({
    method: 'GET',
    url: '/:id'
  }, (error, res) => {
    t.error(error)
    t.equal(res.statusCode, 200)
  })
```

Thank you!

Reply:
I've used the second approach.